File: antilavado-service/src/main/java/ar/com/jf/antilavado/service/common/CommonService.java
package ar.com.jf.antilavado.service.common;

import ar.com.jf.antilavado.repository.contant.StatusTypes;
import ar.com.jf.antilavado.repository.dto.response.CodDesc;
import ar.com.jf.antilavado.repository.model.*;

import java.util.List;

/**
 * CommonService.java
 * <p>
 * PLAYFT.
 * <p>
 * Copyright (c) 2015 ****Fernando Valdes <fervaldes11@gmail.com>****
 * <p>
 * Created by fvaldes on 14/09/2015.
 */
public interface CommonService {

    List<Bank> banks();

    List<AccountType> accountsType();

    List<DocumentType> documentsType();

    List<TelephoneType> telephonesType();

    List<Country> countries();

    List<Province> provinces(Long countryId);

    List<District> districtsBy(Long provinceId);

    List<Locality> localitiesBy(Long localitiesId);

    List<AFIPActivity> parentAFIPActivities();

    List<AFIPActivity> childrenAFIPActivities(String parent);

    List<CodDesc> tableNames();

    List<CodDesc> columnNames(String tableName) throws Exception;

    List<StatusType> statusTypes(StatusTypes statusType);

    List<ValuesType> valuesTypes(Long parentId);
}

File: antilavado-web/src/main/webapp/modules/core/services/common.service.js
/**
 * common.service.js
 *
 * PLAYFT.
 *
 * Copyright (c) 2015 ****Fernando Valdes <fervaldes11@gmail.com>****
 *
 * Created by ktufernando on 10/11/2015.
 */

'use strict';

angular.module(ApplicationConfiguration.applicationModuleName).factory('CommonService', function ($resource, BASE_CONFIG) {
    return {
        banks : $resource(BASE_CONFIG.basePath + 'user/common/banks'),
        accountsTypes : $resource(BASE_CONFIG.basePath + 'user/common/accountsTypes'),
        documentsType : $resource(BASE_CONFIG.basePath + 'user/common/documentsType'),
        taxpayersType : $resource(BASE_CONFIG.basePath + 'user/common/taxpayersType'),
        telephonesType : $resource(BASE_CONFIG.basePath + 'user/common/telephonesType'),
        countries : $resource(BASE_CONFIG.basePath + 'user/common/countries'),
        provinces : $resource(BASE_CONFIG.basePath + 'user/common/:countryId/provinces'),
        districts : $resource(BASE_CONFIG.basePath + 'user/common/:provinceId/districts'),
        localities : $resource(BASE_CONFIG.basePath + 'user/common/:districtId/localities'),
        parentAFIPActivities : $resource(BASE_CONFIG.basePath + 'user/common/parentAFIPActivities'),
        childrenAFIPActivities : $resource(BASE_CONFIG.basePath + 'user/common/childrenAFIPActivities/:parent'),
        marketsBackground : $resource(BASE_CONFIG.basePath + 'user/common/marketsBackground'),
        statusTypes : $resource(BASE_CONFIG.basePath + 'user/common/statusTypes/:type'),
        valuesTypes : $resource(BASE_CONFIG.basePath + 'user/common/valuesTypes/:parentId'),
        /*tableNames : $resource(BASE_CONFIG.basePath + 'user/common/tableNames'),
        columnNames : $resource(BASE_CONFIG.basePath + 'user/common/columnNames/:tableName'),*/
        sino : [{value:true, description:"SI"}, {value:false, description:"NO"}],
        status : [{value:1, description:"ACTIVO"}, {value:0, description:"INACTIVO"}],
        frequencyLabel : $resource(BASE_CONFIG.basePath + 'user/common/frequency/label')
    }
});


